internal/datastore/filetracker/filetracker.go

```diff
@@ -16,6 +19,8 @@ type FileTracker struct {
 	// Hasher allows to change the way that hashes are calculated. Uses xxHash32Hasher{} by default.
 	// Useful for testing.
 	Hasher Hasher
+
+	logger log.Logger
```

Comment:
This logger could be different from the `app.Logger`. With the current implementation it's going to be the same, but the logger may should be injected. 

This is not something you should address, but it's one of the problems I saw to use loggers inside some parts of the code :-D